apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletTransactionsLimitted.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { useQuery } from '@tanstack/react-query'
+
+import useLimitedWalletAddresses from '@/api/apiDataHooks/utils/useLimitedWalletAddresses'
+import { useFetchWalletLastTransaction } from '@/api/apiDataHooks/wallet/useFetchWalletLastTransactions'
+import { walletLatestTransactionsQuery } from '@/api/queries/transactionQueries'
+import { useAppSelector } from '@/hooks/redux'
+
+const useFetchWalletTransactionsLimitted = () => {
+  const networkId = useAppSelector((s) => s.network.settings.networkId)
+  const { addressHashes, isLimited } = useLimitedWalletAddresses()
```

Comment:
Here again, I would avoid to have a specific `useFetchWalletTransactionsLimited` hook.
I'd rather have a single `useFetchWalletLatestTransactions` which returns a `isLimitedByAddressListSize` boolean, based on the returned value of a single `useFetchWalletAddresses` (which would return the `isLimited` boolean).

Reply:
Are you proposing of merging `useFetchWalletTransactionsLimited` and `useFetchWalletTransactionsInfinite`? Or are you proposing remaining this hook from `useFetchWalletTransactionsLimitted` to `useFetchWalletLatestTransactions`?

If it's the first case, that would result to a very complicated hook because one needs to use `useQuery` and the other `useInfiniteQuery` with much different logic for handling refreshing.

If it's the second case, I don't mind renaming, but it makes sense to me to have the prefix `Limited` and `Infinite`.

Is there a third case?

Reply:
Much better after the renames we talked about!